src/streamlink_cli/utils/progress.py

```diff
@@ -13,22 +13,51 @@
     "[download] {written}"
 )
 
+# widths generated from
+# http://www.unicode.org/Public/4.0-Update/EastAsianWidth-4.0.0.txt
 
-def terminal_len(value):
-    """Returns the length of the string it would be when displayed.
 
-    Attempts to decode the string as UTF-8 first if it's a bytestring.
-    """
+widths = [
```

Comment:
`unicodedata.east_asian_width` might be a better way to get this data?

Reply:
https://stackoverflow.com/questions/48598304/width-of-a-string-with-zero-width-and-two-width-characters-in-python-3-in-a-ter

Reply:
Using `east_asian_width` has some issues. (https://stackoverflow.com/a/31666966)

The width table in this Pull Request is based on https://github.com/urwid/urwid/blob/master/urwid/old_str_util.py#L38

Another better way is using library `wcwidth`(https://github.com/jquast/wcwidth)

Thanks for reviewing.